src/core/qgsmaphittest.cpp

```diff
@@ -205,3 +206,17 @@ void QgsMapHitTest::runHitTestLayer( QgsVectorLayer *vl, SymbolSet &usedSymbols,
   }
 }
 
+bool QgsMapHitTest::rasterVisible( QgsRasterLayer *layer ) const
+{
+  if ( ! layer->dataProvider() )
+    return false;
+  QgsRectangle footprint = layer->dataProvider()->extent();
+  if ( mSettings.destinationCrs() != layer->crs() )
+  {
+    QgsCoordinateTransform ct = QgsCoordinateTransform( mSettings.destinationCrs(), layer->crs(), mSettings.transformContext() );
+    footprint = ct.transformBoundingBox( footprint );
```

Comment:
Is it not possible for this to throw a `QgsCsException` that should be handled?

Reply:
Yes. though it's not currently done for the vector hit test.